spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1388,6 +1388,19 @@ def compute_whitespace(line):
     @request(method=LSPRequestTypes.DOCUMENT_FOLDING_RANGE)
     def request_folding(self):
         """Request folding."""
+        total_lines = self.get_line_count()
+        if total_lines > 2000 and self.code_folding:
+            warn_seen = CONF.get('editor', 'code_folding_warn')
+            warn_str = _('This file contains more than 2000 lines! All '
+                         'code folding functionality will be disabled in '
+                         'order to prevent further performance degradation.')
+            if not warn_seen:

```

Comment:
```suggestion
            if not warn:
```

Reply:
Shouldn't this be: ``if warn``?

Reply:
Sorry, you're totally right! Else it wouldn't be shown. Fixed in my suggestion above.